File: CPlusPlus/RPNParser/Node.cpp
#ifndef Node_CPP
#define Node_CPP

#include <iostream>
#include <cmath>

class Node{
    public:
        Node()
        {
            value = 0;
            next = NULL;
        }
        Node(double a)
        {
            value = a;
        }
        double getValue()
        {
            return value;
        }


        Node getNext()
        {
            return *next;
        }
        void setNext(Node* n)
        {
            next = n;
        }
    private:
        double value;
        Node *next;
};
#endif

File: Java/RPNParser/Node.java
import java.io.*;

public class Node
{
    private double data;
    private Node next;

    // region Constructors
    public Node()
    {
        data = 0;
    }
    public Node(double temp)
    {
        data = temp;
    }
    // endregion

    // region Pointers
    public Node getNext()
    {
        return next;
    }
    public void setNext(Node nextNode)
    {
        next = nextNode;
    }
    // endregion

    // region Data
    public double getData()
    {
        return data;
    }
    // endregion
}